packages/neos-ui/src/Containers/ContentCanvas/index.js

```diff
@@ -117,105 +100,10 @@ export default class ContentCanvas extends PureComponent {
-        const nodes = iframeWindow['@Neos.Neos.Ui:Nodes'];
-        nodes[documentInformation.metaData.contextPath] = documentInformation.metaData.documentNodeSerialization;
-        addNodes(nodes);
-
-        setContextPath(documentInformation.metaData.contextPath, documentInformation.metaData.siteNode);
-        setPreviewUrl(documentInformation.metaData.previewUrl);
-        setActiveDimensions(documentInformation.metaData.contentDimensions.active);
-
-        // WARNING: we need to initialize byContextPathDynamicAccess AFTER addNodes from above; otherwise the access function
-        // will be based on the OLD state, and thus NOT return the new nodes...
-        const {
-            focusNode,
-            formattingUnderCursor,
-            setCurrentlyEditedPropertyName,
-            unFocusNode,
-            byContextPathDynamicAccess,
-            persistChange,
-            formattingRulesRegistry,
-            globalRegistry
-        } = this.props;
-
-        //
-        // Initialize node components
-        //
-        const components = iframeDocument.querySelectorAll('[data-__neos-node-contextpath]');
-        Array.prototype.forEach.call(components, node => {
-            const contextPath = node.getAttribute('data-__neos-node-contextpath');
-            const isHidden = $get([contextPath, 'properties', '_hidden'], nodes);
-
-            if (isHidden) {
-                node.classList.add(style.markHiddenNodeAsHidden);
-            }
-
-            initializeHoverHandlersInIFrame(node, iframeDocument);
-        });
-
-        //
-        // Initialize click outside handler
-        //
-        iframeDocument.body.addEventListener('click', e => {
-            const clickPath = Array.prototype.slice.call(e.path);
-            const isInsideInlineUi = clickPath.filter(domNode =>
-                domNode &&
-                domNode.getAttribute &&
-                domNode.getAttribute('data-__neos__inlineUI')
-            ).length > 0;
+        const {documentInitialized} = this.props;
 
-            const selectedDomNode = clickPath.find(domNode => domNode && domNode.getAttribute && domNode.getAttribute('data-__neos-node-contextpath'));
-
-            if (isInsideInlineUi) {
-                // Do nothing, everything OK!
-            } else if (selectedDomNode) {
-                const contextPath = selectedDomNode.getAttribute('data-__neos-node-contextpath');
-                const fusionPath = selectedDomNode.getAttribute('data-__neos-fusion-path');
-
-                focusNode(contextPath, fusionPath);
-            } else {
-                unFocusNode();
-            }
-        });
-
-        const editorConfig = {
-            formattingRules: formattingRulesRegistry.getAllAsObject(),
-            setFormattingUnderCursor: formattingUnderCursor,
-            setCurrentlyEditedPropertyName
-        };
-
-        iframeWindow.NeosCKEditorApi.initialize(editorConfig);
+        iframeDocument.__isInitialized = true;
```

Comment:
it's used about six lines above to prevent the initialization from occurring multiple times ;)